artifact/parser.cpp

```diff
@@ -45,6 +45,35 @@ namespace version = mender::artifact::v3::version;
 namespace manifest = mender::artifact::v3::manifest;
 namespace payload = mender::artifact::v3::payload;
 
+ExpectedArtifact VerifyEmptyPayloadArtifact(
+	Artifact &artifact, lexer::Lexer<token::Token, token::Type> &lexer) {
+	// No meta-data allowed
+	if (artifact.header.subHeaders.at(0).metadata) {
+		return expected::unexpected(parser_error::MakeError(
+			parser_error::Code::ParseError,
+			"Empty payload Artifacts cannot contain a meta-data section"));
+	}
+	// TODO - When augmented sections are added - check for these also
+	log::Trace("Empty payload Artifact: Verifying empty payload");
+	auto tok = lexer.Next();
+	if (tok.type == token::Type::Payload) {
+		// If a payload is present, verify that it is empty
+		auto expected_payload = artifact.Next();
+		if (!expected_payload) {
+			return expected::unexpected(parser_error::MakeError(
+				parser_error::Code::ParseError,
+				"This should never happen, we have a payload token / programmer error"));
+		}
+		auto payload = expected_payload.value();
+		auto expected_payload_file = payload.Next();
+		if (expected_payload_file) {
+			return expected::unexpected(parser_error::MakeError(
+				parser_error::Code::ParseError, "Empty Payload Artifacts cannot have a payload"));
+		}
```

Comment:
Should you check that the error is `NoMorePayloadFilesError` and not some other error?